builtin/common/settings/dlg_change_mapgen_flags.lua

```diff
@@ -46,33 +46,54 @@ local function get_formspec(dialogdata)
 	end
 
 	local fields = {}
-	local function add_field(x, name, label, value)
+	local function add_field(x, name, label, value, tooltips)
 		fields[#fields + 1] = ("field[%f,%f;3.3,1;%s;%s;%s]"):format(
 			x, height, name, label, core.formspec_escape(value or "")
 		)
+		if tooltips then
+			fields[#fields + 1] = ("tooltip[%s;%s]"):format(name, table.concat(tooltips, "\n"))
+		end
 	end
 	-- First row
 	height = height + 0.3
-	add_field(0.3, "te_offset", fgettext("Offset"), t[1])
-	add_field(3.6, "te_scale",  fgettext("Scale"),  t[2])
-	add_field(6.9, "te_seed",   fgettext("Seed"),   t[6])
+	add_field(0.3, "te_offset", fgettext("Offset"), t[1], {fgettext("This value offsets the final noise.")})
+	add_field(3.6, "te_scale",  fgettext("Scale"),  t[2], {
+			fgettext("This value is multiplied with the noise before the offset is added.")})
+	add_field(6.9, "te_seed",   fgettext("Seed"),   t[6], {
+			fgettext("This value acts as a random seed for the noise. The same seed results in the same noise.")})
 	height = height + 1.1
 
 	-- Second row
-	add_field(0.3, "te_spreadx", fgettext("X spread"), t[3])
+	add_field(0.3, "te_spreadx", fgettext("X spread"), t[3], {
+			fgettext("This value scales the noise in the X axis by this value."),
+			fgettext("This is also the scale of the largest structures in the X direction of the noise.")})
 	if dimension == 3 then
-		add_field(3.6, "te_spready", fgettext("Y spread"), t[4])
+		add_field(3.6, "te_spready", fgettext("Y spread"), t[4], {
+			fgettext("This value scales the noise in the Y axis by this value."),
```

Comment:
I think you could save some translation effort by making the coordinate a placeholder (`$1`).

-> `fgettext("foo $1 axis bar", "X")`